crates/goose/src/agents/recipe_tools/sub_recipe_tools.rs

```diff
@@ -54,27 +56,39 @@ fn extract_task_parameters(params: &Value) -> Vec<Value> {
 fn create_tasks_from_params(
     sub_recipe: &SubRecipe,
     command_params: &[std::collections::HashMap<String, String>],
-) -> Vec<Task> {
-    let tasks: Vec<Task> = command_params
-        .iter()
-        .map(|task_command_param| {
-            let payload = json!({
-                "sub_recipe": {
-                    "name": sub_recipe.name.clone(),
-                    "command_parameters": task_command_param,
-                    "recipe_path": sub_recipe.path.clone(),
-                    "sequential_when_repeated": sub_recipe.sequential_when_repeated
-                }
-            });
-            Task {
-                id: uuid::Uuid::new_v4().to_string(),
-                task_type: TaskType::SubRecipe,
-                payload,
-            }
-        })
-        .collect();
+) -> Result<Vec<Task>> {
+    let mut tasks = Vec::new();
+
+    for task_command_param in command_params {
+        // Load the recipe file for each task (can't clone RecipeFile)
```

Comment:
you can just make the RecipeFile `derive(Clone)`

but even better, it looks like build_recipe_from_template and the functions it calls should just be able to borrow the content anyway, they don't need to take ownership

loading the same file N times is definitely not necessary